### PR TITLE
added feature to view mutual trips in  friends profile page

### LINKED ISSUE #284: Show mutual trips in a friend's profile page
### Steps to reproduce
- Go to `my trips` from navigation drawer
- Select any friend.
- Friend's profile page opens up

### Expected behaviour
- Show a list of user's mutual trips with that friend
- At the bottom, show a heading `Your trips together...`
- Then, show a list of trips
    - Show trip list item similar to that of `my trips`

### Endpoint to use
`get-common-trips/<friends_user_id>`

### PR DESCRIPTION
**Pull request by aayushinigam**
# Description
Fixes #284 
mutualtrips

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Swati4star** [comment]
Can we have a screenshot in which there actually are some mutual trips?

**Comment by Swati4star** [comment]
Don't create a new activity for this, show mutual trips in the firend's profile page only. Similar to what I have written in the issue description.

> ### Expected behaviour
> - Show a list of user's mutual trips with that friend
> - At the bottom, show a heading `Your trips together...`
> - Then, show a list of trips
>     - Show trip list item similar to that of `my trips`
> 

Create a new heading `Your trips together...` & just below that show mutual trips

**Comment by aayushinigam** [comment]
Ah I misunderstood. I thought trips are supposed to be shown as that of myTrips. I'll change that.

**Comment by aayushinigam** [comment]
@Swati4star Done!

**Comment by Swati4star** [comment]
@aayushinigam Do you hide the view in case there are no mutual trips?

**Comment by aayushinigam** [comment]
There will always be one because that's how a friend is added. Through a trip only.  If there is no trip together, friend will not be displayed.

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java [hunk 1]
```diff
@@ -203,6 +191,77 @@ public void onResponse(Call call, final Response response) throws IOException {
...
+                                String image = array.length() > 0 ? array.getString(0) : null;
+                                mTrips.add(new Trip(id, name, image, start, end, tname));
+                                // Specify a layout for RecyclerView
+                                // Create a horizontal RecyclerView
+                                mLayoutManager = new LinearLayoutManager(FriendsProfileActivity.this,
+                                        LinearLayoutManager.HORIZONTAL,
+                                        false
+                                );
+                                recyclerView.setLayoutManager(mLayoutManager);
+                                mMutualTripsAdapter = new MutualTripsAdapter(FriendsProfileActivity.this, mTrips);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java]: You are creating new adapter in the loop. A better option would be just create mTrips array in the loop & set adapter after the loop.

Comment (aayushinigam) [reply Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java]: Resolved.

**Pull request merged**